zkit/src/groth16_api.rs

```diff
@@ -176,39 +178,31 @@ fn read_pk_from_file<E: Engine>(file_path: &str, checked: bool) -> Result<Parame
     Ok(Parameters::<E>::read(&mut reader, checked)?)
 }
 
-fn read_vk_from_file<E: Engine>(file_path: &str) -> Result<VerifyingKey<E>> {
-    let file = std::fs::File::open(file_path)?;
-    let mut reader = std::io::BufReader::new(file);
-    Ok(VerifyingKey::<E>::read(&mut reader)?)
-}
-
-fn read_public_input_from_file(file_path: &str) -> Result<Vec<Fr>> {
-    let file = std::fs::File::open(file_path)?;
-    let mut reader = std::io::BufReader::new(file);
-    Ok(read_fr_vec::<Fr, _>(&mut reader)?)
+fn read_vk_from_file<P: Parser>(file_path: &str) -> Result<VerifyingKey<P>> {
+    let json_data = std::fs::read_to_string(file_path).expect("Unable to read the JSON file");
+    Ok(to_verification_key::<P>(&json_data))
 }
 
-fn read_public_input_from_file_bls12381(file_path: &str) -> Result<Vec<Fr_bls12381>> {
-    let file = std::fs::File::open(file_path)?;
-    let mut reader = std::io::BufReader::new(file);
-    Ok(read_fr_vec::<Fr_bls12381, _>(&mut reader)?)
+fn read_public_input_from_file<T: FieldElement>(file_path: &str) -> Result<Vec<T>> {
+    let json_data = std::fs::read_to_string(file_path).expect("Unable to read the JSON file");
+    Ok(to_public_input::<T>(&json_data))
 }
 
-fn read_proof_from_file<E: Engine>(file_path: &str) -> Result<Proof<E>> {
-    let file = std::fs::File::open(file_path)?;
-    let mut reader = std::io::BufReader::new(file);
-    Ok(Proof::<E>::read(&mut reader)?)
+fn read_proof_from_file<P: Parser>(file_path: &str) -> Result<Proof<P>> {
+    let json_data = std::fs::read_to_string(file_path).expect("Unable to read the JSON file");
+    Ok(to_proof::<P>(&json_data))
```

Comment:
Can we use modifier ? here? If we return a Result, we should not use expect or unwrap as far as we can.